src/components/course_table.rs

```diff
@@ -0,0 +1,67 @@
+use sycamore::prelude::*;
+
+// table structure
+#[component]
+pub fn CourseTable<'a, G: Html>(
+    cx: Scope<'a>,
+) -> View<G> {
+    view! { cx,
+        div (class="overflow-x-auto w-full shadow-md rounded-md") {
+            // TODO: change header color, change hover color
+            table (class="table table-auto table-md w-full") {
+                // table header
+                thead(class="") {
+                    tr() {
+                        th(class="bg-primary"){} // row index
+                        th(class="bg-primary") { p(class="text-base normal-case text-primary-content") {"Code"} }
+                        th(class="bg-primary") { p(class="text-base normal-case text-primary-content") {"Course Name"} }
+                        th(class="bg-primary") { p(class="text-base normal-case text-primary-content") {"Location"} }
+                        th(class="bg-primary") { p(class="text-base normal-case text-primary-content") {"Status"} }
+                    }
+                }
+                // table content
+                tbody() {
+                    TableContent(name = "testing".to_string())
```

Comment:
Could you please add more realistic test data? It should be a mockup that looks like the final website, but just doesn't function. Also, please generate this from a dummy Vector of courses, like search results.

We should also display more than just the name. What should the final UI display? We need a complete mockup!